keps/sig-windows/1122-windows-csi-support/README.md

```diff
@@ -768,12 +777,366 @@ Downgrading the version of csi-proxy.exe to one that is not supported by all ins
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+-->
+
+- [ ] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name:
+  - Components depending on the feature gate:
+- [x] Other
+  - Describe the mechanism:
+  CSI-proxy is a binary that can be deployed (or running as a service) directly on each Windows node during node start up time. For example, in gce, CSI-proxy
+  binary is downloaded and running as Windows service in node [startup script](https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/windows/k8s-node-setup.psm1#L424)
+  An environment variable ENABLE_CSI_PROXY is implemented to disable/enable CSI-proxy installation.
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+  No
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+  If csi-proxy binary is deployed as a binary or windows service during node start up time, enabling this feature will require reprovisioning of a node.
+
+###### Does enabling the feature change any default behavior?
+
+<!--
+Any change of default behavior may be surprising to users or break existing
+automations, so be extremely careful here.
+-->
+No
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+<!--
+Describe the consequences on existing workloads (e.g., if this is a runtime
+feature, can it break the existing applications?).
+
+NOTE: Also set `disable-supported` to `true` or `false` in `kep.yaml`.
+-->
+It depends on cloud provider implementation. For example in GCE, environment variable ENABLE_CSI_PROXY is implemented to disable/enable CSI-proxy installation.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Nothing
+
+###### Are there any tests for feature enablement/disablement?
+
+<!--
+The e2e framework does not currently support enabling or disabling feature
+gates. However, unit tests in each component dealing with managing data, created
+with and without the feature, are necessary. At the very least, think about
+conversion tests if API types are being modified.
+-->
+No
+### Rollout, Upgrade and Rollback Planning
```

Comment:
@ddebroy @jingxu97 how much of the upgrade strategy section has changed? My understanding is now CSI proxy exposes multiple versioned named pipes to facilitate multiple drivers using different API versions, and drivers upgrading to newer API versions.

And on the driver side, @mauriciopoppe did some new experiments so that drivers could fall back to older versions if a newer API version was not available.